cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ProgramLifecycleService.java

```diff
@@ -165,12 +166,24 @@ public ProgramStatus getProgramStatus(ProgramId programId) throws Exception {
       throw new NotFoundException(Ids.namespace(programId.getNamespace()).app(programId.getApplication()).toId());
     }
 
+    return getExistingAppProgramStatus(appSpec, programId);
+  }
+
+  /**
+   * Returns the program status with no need of application existence check.
+   * @param appSpec the ApplicationSpecification of the existing application
+   * @param programId the id of the program for which the status call is made
+   * @return the status of the program
+   * @throws NotFoundException if the application to which this program belongs was not found
+   */
+  private ProgramStatus getExistingAppProgramStatus(ApplicationSpecification appSpec, ProgramId programId)
```

Comment:
this is a private helper method. Does it need to have the same name as the public method?